File: src/client_gen/csharp/Debug.cs
using System.IO;

namespace ProjectName
{
    public class Debug
    {
        private BinaryWriter writer;
        public Debug(BinaryWriter writer)
        {
            this.writer = writer;
        }
        public void Send(Model.DebugCommand command)
        {
            new Model.ClientMessage.DebugMessage(command).WriteTo(writer);
            writer.Flush();
        }
    }
}

File: src/client_gen/cpp/Debug.cpp
#include "Debug.hpp"
#include "model/ClientMessage.hpp"

Debug::Debug(const std::shared_ptr<OutputStream>& outputStream)
    : outputStream(outputStream)
{
}

void Debug::send(const DebugCommand& command)
{
    // TODO: Construct actual message, this is a hack :)
    outputStream->write(ClientMessage::DebugMessage::TAG);
    command.writeTo(*outputStream);
    outputStream->flush();
}